api/app/Http/Controllers/EntityController.php

```diff
@@ -116,30 +131,31 @@ public function putMany(Request $request)
     {
         $requestCollection = $request->data;
 
-        $ids = $this->getIds($requestCollection, $this->getKeyName(), $this->validateRequestRule);
+        $ids = $this->getIds($requestCollection, $this->getModel()->getKeyName(), $this->validateRequestRule);
         $models = [];
         if (!empty($ids)) {
-            $models = $this->getRepository()->findMany($ids);
+            $models = $this->getManyEntitiesByIds($ids);
         }
 
         $putModels = [];
+        $keyName = $this->getModel()->getKeyName();
         foreach ($requestCollection as $requestEntity) {
-            $id = isset($requestEntity[$this->getKeyName()])?$requestEntity[$this->getKeyName()]:null;
+            $id = isset($requestEntity[$keyName])?$requestEntity[$keyName]:null;
```

Comment:
Yes, thats why the `putMany` method would use a `$this->findOrNewCollection($requestCollection)` and the  `patchMany` and `deleteMany` methods would use `$this->findOrFailCollection($requestCollection)`. 

Note the difference with the patch and delete failing on `ModelNotFoundException` but the put returning a new model instead.